### PR TITLE
refactor(libhead): unify VerifyAdjacent and VerifyNonAdjacent into Verify

### PR DESCRIPTION
**Pull request by Wondertan**
Last beautification for read-only header interface. We didn't do it previously with @tzdybal, as the syncing logic depended on adjacency through ErrNotNotAdjacent, which VerifyAdjacent returned. The solution is to keep returning this error but from Append.                                                                                                                                                                                                                                                                   

Additionally, this solves an edge case found by @gupadhyaya, where when Syncer starts and instantly requests and fails to apply a new network header due to adjacency.

### CODE REVIEW

**Comment by Wondertan** [comment]
I also moved /headertest from root to /header. It shouldn't be in the root

**Comment by vgonkivs** [comment]
conflicts

**Comment by Wondertan** [comment]
Lint is borken

**Comment by Wondertan** [comment]
Merge over GH didn't go well, so rebasing locally

File: libs/header/p2p/session.go [hunk 1]
```diff
@@ -201,23 +203,18 @@ func (s *session[H]) validate(headers []H) error {
...
-	for i := 1; i < len(headers); i++ {
-		err = trusted.VerifyAdjacent(headers[i])
+	for i := 0; i < len(headers); i++ {
+		err := trusted.Verify(headers[i])
 		if err != nil {
 			return err
 		}
+		if trusted.Height()+1 != headers[i].Height() {
+			// Exchange requires requested ranges to always consist of adjacent headers
+			return fmt.Errorf("peer sent valid but undjacent header")
```
Comment (renaynay) [inline libs/header/p2p/session.go:215]:
```suggestion
			return fmt.Errorf("peer sent valid but non-adjacent header")
```

File: libs/header/p2p/session.go [hunk 2]
```diff
@@ -201,23 +203,18 @@ func (s *session[H]) validate(headers []H) error {
...
-		return nil
-	}
-
-	trusted := headers[0]
+	trusted := s.from
 	// verify that the whole range is valid.
-	for i := 1; i < len(headers); i++ {
-		err = trusted.VerifyAdjacent(headers[i])
+	for i := 0; i < len(headers); i++ {
+		err := trusted.Verify(headers[i])
```
Comment (renaynay) [inline libs/header/p2p/session.go:209]: if last header passes verification, shouldn't trusted = header[i-1] then or no?

File: libs/header/sync/sync_head.go [hunk 3]
```diff
@@ -183,6 +183,9 @@ func (s *Syncer[H]) validate(ctx context.Context, new H) pubsub.ValidationResult
 	return pubsub.ValidationAccept
 }
 
+// TODO(@Wondertan): We should request TrustingPeriod from the network's state params or
```
Comment (renaynay) [inline libs/header/sync/sync_head.go:186]: can you put in issue?

**Review by renaynay — DISMISSED** [review]

File: libs/header/sync/sync_head.go (hunk 3 above)
Comment (Wondertan) [reply libs/header/sync/sync_head.go:186]: Will revalidate this first

**Review by vgonkivs — APPROVED** [review]

File: libs/header/sync/sync_head.go (hunk 3 above)
Comment (Wondertan) [reply libs/header/sync/sync_head.go:186]: https://github.com/celestiaorg/celestia-node/issues/1857

**Review by renaynay — APPROVED** [review]

**Pull request merged**